examples/decoding/plot_receptive_field.py

```diff
@@ -0,0 +1,132 @@
+can perform a similar function along with :mod:`sklearn`. We will fit a
+linear encoding model using the continuously-varying speech envelope to
+predict activity of a 128 channel EEG system.
+
+References
+----------
+.. [1] Crosse, M. J., Di Liberto, G. M., Bednar, A. & Lalor, E. C. (2016).
+       The Multivariate Temporal Response Function (mTRF) Toolbox:
+       A MATLAB Toolbox for Relating Neural Signals to Continuous Stimuli.
+       Frontiers in Human Neuroscience 10, 604. doi:10.3389/fnhum.2016.00604
+"""
+# Authors: Chris Holdgraf <choldgraf@gmail.com>
+#
+# License: BSD (3-clause)
+
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+import mne
+from mne.decoding import ReceptiveField
+
+from sklearn.linear_model import Ridge
+from sklearn.model_selection import KFold
+from sklearn.preprocessing import scale
+from scipy.io import loadmat
+from scipy.stats import pearsonr
+
+
+###############################################################################
+# Load the data from the publication
+path = mne.datasets.mtrf.data_path()
+n_decim = 2  # To speed up computation
+data = loadmat(path + '/speech_data.mat')
+raw = data['EEG'].T
+speech = data['envelope'].T
+sfreq = float(data['Fs'].squeeze())
+sfreq /= n_decim
+speech = mne.filter.resample(speech, 1, n_decim, npad='auto')
+raw = mne.filter.resample(raw, 1, n_decim, npad='auto')
+
+
+# Read in channel positions and create our MNE objects from the raw data
+montage = mne.channels.read_montage('biosemi128')
+montage.selection = montage.selection[:128]
+layout = mne.channels.read_layout('Vectorview-all')
+info = mne.create_info(montage.ch_names[:128], sfreq, 'eeg', montage=montage)
+layout = mne.channels.make_eeg_layout(info)
+raw = mne.io.RawArray(raw, info)
+
+###############################################################################
+# Plot a sample of brain and stimulus activity
+fig, ax = plt.subplots()
+ax.plot(scale(raw[:][0][..., :800].T), color='k', alpha=.1)
+ax.plot(scale(speech[0, :800]), color='r', lw=2)
+ax.set(title="Sample brain and stimulus activity", xlabel="Time (s)")
+mne.viz.tight_layout()
+
+###############################################################################
+# Create and fit a receptive field model
+
+# Define the delays that we will use in the receptive field
+tmin, tmax = -.4, .2
+
+# Initialize the model
+spec_names = ['freq_%s' % ii for ii in range(speech.shape[0])]
+rf = ReceptiveField(tmin, tmax, sfreq,
+                    feature_names=spec_names, model=Ridge())
+n_delays = int((tmax - tmin) * sfreq) + 2  # +2 to account for 0 + end
+
+n_splits = 3
+cv = KFold(n_splits)
+
+Y = raw[:][0]  # Outputs for the model
+
+# Iterate through folds, fit the model, and predict/test on held-out data
+coefs = np.zeros([n_splits, len(raw.ch_names), n_delays])
+scores = np.zeros([n_splits, len(raw.ch_names)])
+for ii, (train, test) in enumerate(cv.split(speech.T)):
+    print('CV iteration %s' % ii)
+    rf.fit(speech[..., train], Y[..., train])
```

Comment:
re: input shape:  I had thought that in a previous iteration of this PR people said we shouldn't worry about explicit sklearn API right now and instead use a standard MNE input shape.

re: CV: same goes for this point. In the last (`EncodingModel`) API, I built CV etc into the object itself, and then I'm pretty sure folks said that it was too complex and that we shouldn't worry about supporting CV within the object. Maybe I am mis-remembering.